src/server/services/wfs3/qgswfs3handlers.cpp

```diff
@@ -442,9 +442,10 @@ void QgsWfs3DescribeCollectionHandler::handleRequest( const QgsServerApiContext
   const std::string title { mapLayer->serverProperties()->wfsTitle().isEmpty() ? mapLayer->name().toStdString() : mapLayer->serverProperties()->wfsTitle().toStdString() };
   const std::string itemsTitle { title + " items" };
   const QString shortName { mapLayer->serverProperties()->shortName().isEmpty() ? mapLayer->name() : mapLayer->serverProperties()->shortName() };
+  const QString typeName = QString( shortName ).replace( ' ', '_' ).replace( ':', '-' ).replace( QChar( 0x2014 ) /* em-dash */, '-' );
```

Comment:
Can't you just use QString layerTypeName( const QgsMapLayer *layer ) from wfsutils?

Reply:
`wfsutils.cpp` is not in the `wfs` service module, so using it in the `wfs3` service module will result in an undefined reference. Moving `wfsutils.cpp` to the qgis_server compilation unit is somewhat invasive. An alternative could be to move `layerTypeName` to `QgsMapLayerServerProperties`?

Reply:
That works, I am just trying to avoid code duplication.